File: 02-query/node/getRoles.js
const AWS = require("aws-sdk");
const DynamoDB = new AWS.DynamoDB({ region: "us-east-1" });

DynamoDB.batchGetItem(
  {
    RequestItems: {
      MovieRoles: {
        Keys: [
          { Actor: { S: "Tom Hanks" }, Movie: { S: "Cast Away" } },
          { Actor: { S: "Tom Hanks" }, Movie: { S: "Toy Story" } },
          { Actor: { S: "Tim Allen" }, Movie: { S: "Toy Story" } },
          { Actor: { S: "Natalie Portman" }, Movie: { S: "Black Swan" } },
        ],
      },
    },
  },
  function (err, data) {
    if (err) console.log(`Error fetching roles": ${err}`);
    else if (!data.Responses) console.log("Roles do not exist");
    else console.log(`Found roles: ${JSON.stringify(data.Responses, null, 2)}`);
  }
);

File: 02-query/python/get_roles.py
import json

import boto3

client = boto3.client("dynamodb", region_name="us-east-1")

try:
    response = client.batch_get_item(
        RequestItems={
            "MovieRoles": {
                "Keys": [
                    {"Actor": {"S": "Tom Hanks"}, "Movie": {"S": "Cast Away"}},
                    {"Actor": {"S": "Tom Hanks"}, "Movie": {"S": "Toy Story"}},
                    {"Actor": {"S": "Tim Allen"}, "Movie": {"S": "Toy Story"}},
                    {
                        "Actor": {"S": "Natalie Portman"},
                        "Movie": {"S": "Black Swan"},
                    },
                ]
            }
        }
    )
    if not response["Responses"]:
        print("Roles do not exist")
    else:
        print(f"Found roles: {json.dumps(response['Responses'], indent=2)}")
except Exception as err:
    print(f"Error fetching roles: {err}")
